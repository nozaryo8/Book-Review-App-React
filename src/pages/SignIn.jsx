import React, { useState } from "react";
import axios from "axios";
// import { useCookies } from "react-cookie";
import { Navigate, useNavigate, Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
// import { signIn } from "../authSlice";
// import { url } from "../const";
import { url } from "../const";

export const SignIn = () => {
    //   const auth = useSelector((state) => state.auth.isSignIn);
    //   const dispatch = useDispatch();
    //   const history = useNavigate();
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [errorMessage, setErrorMessage] = useState();
    //   const [cookies, setCookie, removeCookie] = useCookies();
    const handleEmailChange = (e) => setEmail(e.target.value);
    const handlePasswordChange = (e) => setPassword(e.target.value);
    const onSignIn = () => {
        console.log(process.env);
        console.log(url);
        if(checkForm(email, password)) {
            axios
            .post(`https://ifrbzeaz2b.execute-api.ap-northeast-1.amazonaws.com/signin`, { email: email, password: password })
            .then((res) => {
                // setCookie("token", res.data.token);
                // dispatch(signIn());
                // history('/');
            })
            .catch((err) => {
                setErrorMessage(`サインインに失敗しました。${err}`);
            });
        }
    };
    const checkForm = (email, password) => {
        const pattern = /^[A-Za-z0-9]{1}[A-Za-z0-9_.-]*@{1}[A-Za-z0-9_.-]+.[A-Za-z0-9]+$/;

        if (!pattern.test(email)) {
            setErrorMessage("正式なメールアドレスではありません");
            return false;
        }
        return true;
    }

    //   if (auth) return <Navigate to="/" />;
    return (
        <div>
            <main className="signin">
                <h2>サインイン</h2>
                <p className="error-message">{errorMessage}</p>
                <form className="signin-form">
                    <label className="email-label">メールアドレス</label>
                    <br />
                    <input
                        id="input-email"
                        type="email"
                        className="email-input"
                        onChange={handleEmailChange}
                    />
                    <br />
                    <label className="password-label">パスワード</label>
                    <br />
                    <input
                    type="password"
                    className="password-input"
                    onChange={handlePasswordChange}
                    />
                    <br />
                    <button type="button" className="signin-button" id="submit" onClick={onSignIn}>
                        サインイン
                    </button>
                </form>
                {/* <Link to="/signup">新規作成</Link> */}
            </main>
        </div>
    );
};
