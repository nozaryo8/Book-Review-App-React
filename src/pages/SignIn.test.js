import TestRenderer from 'react-test-renderer';
import { SignIn } from "./SignIn";
import { render, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

// console.log(testRenderer.toJSON())

// // 入力エリア
// const input = testRenderer.getInstance("input-email")

// // デフォルトは空であることを確認
// expect(input).toBe(null)

describe("<SignIn />", () => {
    it("スナップショットテスト", () => {
        const testRenderer = TestRenderer.create(<SignIn />);
        // console.log(testRenderer.toJSON())
        // console.log(testRenderer)
        // console.log(testRenderer.toTree())
        // console.log(testRenderer.root)
        // console.log(testRenderer.root.children)

        // .snapファイルの内容と実際の実装を比較する
        expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it("テキストボックス入力テスト", () => {
        render(<SignIn />);
        const inputValue = screen.getByRole("textbox");
        console.log(screen)
        userEvent.type(inputValue, "test");
        expect(inputValue.value).toBe("test");
    });


})
