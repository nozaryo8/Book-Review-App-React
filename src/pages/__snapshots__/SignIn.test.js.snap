// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> スナップショットテスト 1`] = `
<div>
  <main
    className="signin"
  >
    <h2>
      サインイン
    </h2>
    <p
      className="error-message"
    />
    <form
      className="signin-form"
    >
      <label
        className="email-label"
      >
        メールアドレス
      </label>
      <br />
      <input
        className="email-input"
        id="input-email"
        onChange={[Function]}
        type="email"
      />
      <br />
      <label
        className="password-label"
      >
        パスワード
      </label>
      <br />
      <input
        className="password-input"
        onChange={[Function]}
        type="password"
      />
      <br />
      <button
        className="signin-button"
        id="submit"
        onClick={[Function]}
        type="button"
      >
        サインイン
      </button>
    </form>
  </main>
</div>
`;
